### DISCUSSION TITLE
Auto apply shipping service to cart via API / Show custom field on Shipping Address in Cart UI

### DISCUSSION THREAD

**Discussion by balanh-development** [swellstores/community#242] [Help] [answered]
1. Is there any API (Rest, JS) auto apply shipping service (meet the conditions) on cart without having to choose from customers.
Use case: We have a fixed shipping fee for each region / state, customers do not need to choose the shipping service, but we have to show the price of shipping to customers in cart before it converting it to an order.

2. In Cart UI, we have some custom fields on Shipping address (storage works perfectly), but I can not see any options to display custom files on UI (SHIPPING ADDRESS zone) or in edit tab.

**Comment by sags95** [comment, marked as answer]
Hey @balanh-development ,

1) There's not currently a way for a cart to auto-apply a shipping service, but one option would be to use the Order Webhook that can fire on shipping events. You can then have some logic in this webhook listener to fetch/apply shipping methods or ratings. This flow might  differ depending on if you're using a custom or hosted checkout.

An example of responding to the order webhook and setting the shipping service would be the following:
```
$respond({
  status: 200,
  body: {
    shipping: {
     service:  <id>
   }
  }
})
```

2) Could you maybe share a screenshot of the issue you're experiencing?

**Reply by balanh-development** [reply]
Thanks for your reply,

<img width="312" alt="image" src="https://user-images.githubusercontent.com/120219638/236979394-6262fcd6-55f4-4fb1-bd84-cab9d06f85a6.png">
admin/cart/id -> "SHIPPING ADDRESS"

Our custom field
<img width="1209" alt="image" src="https://user-images.githubusercontent.com/120219638/236979526-46a93ed8-3676-40e9-ac35-af0057cc0731.png">

I want to display my custom fields on Shipping Address Tab (also in Edit Tab), right now it display in cart content below.

<img width="898" alt="image" src="https://user-images.githubusercontent.com/120219638/236979666-4d00b145-5de8-4e38-b607-dee62f7660ae.png">

**Reply by logeshswell** [reply]
Yeah, this functionality is currently not supported. I will make sure to create a feature request for the same.
